wai-logger/Network/Wai/Logger/Apache.hs

```diff
@@ -37,6 +38,8 @@ data IPAddrSource =
     FromSocket
   -- | From X-Real-IP: or X-Forwarded-For: in the HTTP header.
   | FromHeader
+  -- | From a custom HTTP header, useful in proxied environment.
+  | FromHeaderCustom [HeaderName]
```

Comment:
Since `IPAddrSource` is exposed to the user, I believe the addition of a data constructor should have caused a MAJOR version bump, at least according to the [PVP](https://pvp.haskell.org/).

So the new version should have become 2.5, not 2.4.1.

(Apologies for the noise if `wai-logger` is not actually following the PVP on this!)

Reply:
Ah I think you're right, while this is not a breaking change by intention, the compiler should complain about a missing pattern matching case, which could result in a build failure.

Reply:
Ah yeah, just to be clear, in point 1 in https://pvp.haskell.org/#version-numbers, it says:

> Breaking change. If any entity was removed, or the types of any entities or the _definitions of datatypes_ or classes were changed, or orphan instances were added or any instances were removed, then _the new A.B MUST be greater than the previous A.B_.

(emphasis mine)

Here, `A.B.` is referring to the MAJOR version.  (I'm not sure whether these libraries are following PVP or semver.)

Reply:
Right, it's a common pitfall and some library authors choose to hide their constructors and instead expose `set` functions, which is a bit annoying but helps to avoid a major version bump.

Reply:
So, should I deprecate 2.4.1 on Hackage and release it as 2.5?

Reply:
@kazu-yamamoto I believe that is correct.